### PR TITLE
Add client TLS configuration

### PR DESCRIPTION
**Pull request by frebib**
Allows connecting to alertmanager instances behind a TLS endpoint that
requires mutual TLS. Conveniently also allows specifying a CA
certificate file for alertmanagers that use trusted roots not in the
system root trust store.

Fixes: https://github.com/prometheus/alertmanager/issues/2652
Signed-off-by: Joe Groocock <me@frebib.net>

### CODE REVIEW

**Comment by roidelapluie** [comment]
This looks great! Just a few comments.

**Comment by frebib** [comment]
Applied suggestions, squashed and rebased against `main`

**Comment by frebib** [comment]
Build failure is transient. Please re-run 🙏🏻

**Review by roidelapluie — APPROVED** [review]

File: cli/root.go [hunk 1]
```diff
@@ -127,12 +126,20 @@ func Execute() {
 	app.Flag("alertmanager.url", "Alertmanager to talk to").URLVar(&alertmanagerURL)
 	app.Flag("output", "Output formatter (simple, extended, json)").Short('o').Default("simple").EnumVar(&output, "simple", "extended", "json")
 	app.Flag("timeout", "Timeout for the executed command").Default("30s").DurationVar(&timeout)
-	app.Flag("tls.insecure.skip.verify", "Skip TLS certificate verification").BoolVar(&tlsInsecureSkipVerify)
+	app.Flag("tls.certfile", "TLS client certificate file").PlaceHolder("client.crt").ExistingFileVar(&tls.CertFile)
+	app.Flag("tls.keyfile", "TLS client private key file").PlaceHolder("client.key").ExistingFileVar(&tls.KeyFile)
+	app.Flag("tls.cafile", "TLS trusted certificate authorities file").PlaceHolder("ca.crt").ExistingFileVar(&tls.CAFile)
```
Comment (roidelapluie) [inline cli/root.go:131]: Can you remove the files placeholders? They look like default values in --help, I don't like it.

Comment (frebib) [reply cli/root.go:131]: Sure. I wasn't sure what to do with these. I didn't really like it either way. The placeholders shout more, but at least they don't look like filenames

Comment (roidelapluie) [reply cli/root.go:131]: Oh
I have inspiration

Comment (roidelapluie) [reply cli/root.go:131]: In prometheus, the placeholders are like `<url>`, `<path>` (url path), we could do `<file>` for the files. And fix the URL.

Comment (roidelapluie) [reply cli/root.go:131]: Let's take the placeholders from https://prometheus.io/docs/prometheus/latest/configuration/configuration/

<filename> and <string> for the servername.

**Review by roidelapluie — CHANGES_REQUESTED** [review]
Removing my approval until placeholders are fixed. Thanks again for your work

**Comment by roidelapluie** [comment]
(note: we are cutting a release next week)

**Comment by frebib** [comment]
```
$ ./amtool --help | grep -C2 -- --tls\.         
  -o, --output=simple            Output formatter (simple, extended, json)
      --timeout=30s              Timeout for the executed command
      --tls.certfile=<filename>  TLS client certificate file
      --tls.keyfile=<filename>   TLS client private key file
      --tls.cafile=<filename>    TLS trusted certificate authorities file
      --tls.servername=<string>  ServerName to verify hostname of alertmanager
      --tls.insecure.skip.verify
                                 Skip TLS certificate verification
      --version-check            Check alertmanager version. Use
```

**Comment by frebib** [comment]
CI failures seem unrelated again. Re-run?

**Review by roidelapluie — APPROVED** [review]

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**